pkg/cluster/docker/up.go

```diff
@@ -0,0 +1,625 @@
+package docker
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/blang/semver"
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kclientcmd "k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cluster/docker/dockerhelper"
+	"github.com/openshift/origin/pkg/cluster/docker/dockermachine"
+	"github.com/openshift/origin/pkg/cluster/docker/errors"
+	"github.com/openshift/origin/pkg/cluster/docker/host"
+	"github.com/openshift/origin/pkg/cluster/docker/openshift"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	dockerutil "github.com/openshift/origin/pkg/cmd/util/docker"
+)
+
+const (
+	// RecommendedName is the recommended command name
+	RecommendedName = "docker"
+
+	openShiftContainer = "origin"
+	openShiftNamespace = "openshift"
+
+	initialUser     = "developer"
+	initialPassword = "developer"
+
+	initialProjectName    = "myproject"
+	initialProjectDisplay = "My Project"
+	initialProjectDesc    = "Initial developer project"
+
+	cmdUpLong = `
+Starts an OpenShift cluster using Docker containers, provisioning a registry, router,
+initial templates, and a default project.
+
+This command will attempt to use an existing connection to a Docker daemon. Before running 
+the command, ensure that you can execure docker commands successfully (ie. 'docker ps').
+
+Optionally, the command can create a new Docker machine for OpenShift using the VirtualBox
+driver when the --create-machine argument is specified. The machine will be named 'openshift'
+by default. To name the machine differently, use the --docker-machine=NAME argument. If the
+--docker-machine=NAME argument is specified, but --create-machine is not, the command will attempt
+to find an existing docker machine with that name and start it if it's not running.
+
+By default, the OpenShift cluster will be setup with a public address that ends in xip.io.
+This is to allow dynamic host names to be created for routes. An alternate public hostname can
+be specified with the --public-hostname=NAME argument.
+
+`
+	cmdUpExample = `
+# Start OpenShift on a new docker machine named 'openshift'
+$ %[1]s --create-machine
```

Comment:
https://github.com/openshift/origin/pull/8454 landed today which means we won't use prompts anymore.